acceptancetests/assess_persistent_storage.py

```diff
@@ -120,29 +110,120 @@ def assess_charm_deploy_and_remove_single_storage(client):
+            '- {} with {}, test terminated.'.format(
             pool_storage, pool_setting))
             sys.exit(1)
+        else:
+            log.info('Check properties of single block device storage unit: PASSED.')
+
+        # check type, persistent setting and pool of single filesystem storage unit
+        storage_type = storage_output['storage'][single_fs_id]['kind']
+        persistent_setting = storage_output['storage'][single_fs_id]['persistent']
+        pool_storage = storage_output['filesystems']['0/0']['storage']
+        pool_setting = storage_output['filesystems']['0/0']['pool']
+
+        if storage_type != 'filesystem':
+            log.error(
+                'Incorrect type for single filesystem storage detected '\
+                '- {}, test terminated.'.format(storage_type))
+            sys.exit(1)
+        elif persistent_setting != False:
+            log.error(
+                'Incorrect value for persistent setting on single filesystem '\
+                'storage - {}, test terminated.'.format(persistent_setting))
+            sys.exit(1)
+        elif (pool_storage != single_fs_id) and (pool_setting != 'rootfs'):
+            log.error('Incorrect filesystems detected '\
+            '- {} with {}, test terminated.'.format(
+            pool_storage, pool_setting))
+            sys.exit(1)
+        else:
+            log.info('Check properties of single filesystem storage unit: PASSED.')
+    return (single_fs_id, single_blk_id)
 
 
+def assess_charm_removal_single_block_and_filesystem_storage(client):
+    """This function tests charm removal while a single filesystem storage
+       and a single persistent block device storage attached.
+
+       Steps taken to the test:
+       - Run assess_charm_deploy_single_block_and_filesystem_storage(client)
+       - Remove dummy-storage charm while a single block storage unit (ebs)
+         and a single filesystem storage unit (rootfs) attached.
+           > The output should states that there is a persistent storage unit.
+           > The application should be removed successfully.
+       - Check charm storage units once the removal is done.
+           > The filesystem storage unit (rootfs)should be removed successfully.
+           > The block device storage unit (ebs) should remain and detached,
+
+       :param client: ModelClient object to deploy the charm on.
+    """
+
+    charm_name = 'dummy-storage'
+    single_fs_id, single_blk_id = \
+        assess_charm_deploy_single_block_and_filesystem_storage(client)
+    # Run juju remove-application dummy-storage
+    app_removal_output = client.get_juju_output(
+        'remove-application', charm_name, '--show-log', include_e=False, merge_stderr=True)
+    ipdb.set_trace()
+    # pre-set the result to FAILED
+    remove_app_output_check = 'FAILED'
+    remove_single_fs_output_check = 'FAILED'
+    detach_single_blk_output_check = 'FAILED'
+
+    for line in app_removal_output.split('\n'):
```

Comment:
This is a requirement in the spec. The requirement will check 2 things in this case, the first one is to check if the messages in stdout indicate the difference between non-persistent storage (rootfs) and the persistent one (ebs). This is important because it's the first time user sees such difference in comparison on Juju without persistent feature. The second thing is to check if the non-persistent one is removed and persistent one is detached by run juju storage later on, which I haven't done it yet. So, the output string itself is the feature here, it has to be checked, and if that output changed, then yes, we should have a failure, investigate if it's a bug or just a word change but with same meaning. I've tried to keep the required words minimal in the string checking.